File: Code Forces/CPP/div_eight.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
	string num;
	cin>>num;
	num = "00"+num;

	int len = num.length();
	for (int i = 0; i < len-2; ++i)
		for (int j = i+1; j < len-1; ++j)
			for (int k = j+1; k < len; ++k){
				int val = 0;
				val += (num[i]-'0') * 100;
				val += (num[j]-'0') * 10;
				val += (num[k]-'0') * 1;
				if(val%8 == 0){
					cout<<"YES\n"<<val<<endl;
					return 0;
				}
			}

	cout<<"NO"<<endl;
	return 0;
}

File: Code Forces/PY/div_eight.py
num = input()
num = "00"+num

for i in range(len(num)-2):
	for j in range(i+1, len(num)-1):
		for k in range(j+1, len(num)):
			val = int(num[i]+num[j]+num[k])
			if val%8 == 0:
				print("YES")
				print(val)
				exit()

print("NO")

